vapi/simulator/simulator.go

```diff
@@ -156,7 +156,9 @@ func New(u *url.URL, settings []vim.BaseOptionValue) (string, http.Handler) {
 }
 
 func (s *handler) isAuthorized(r *http.Request) bool {
-	if r.Method == http.MethodPost && strings.HasSuffix(r.URL.Path, internal.SessionPath) {
+	if r.Method == http.MethodPost &&
+		strings.HasSuffix(r.URL.Path, internal.SessionPath) &&
+		r.URL.Query().Get(internal.SessionGetParams.Key) == "" {
```

Comment:
You can use the `s.action(r)` method here.

See: https://github.com/vmware/govmomi/blob/ee35ba9e79122df9873f2dc948bcce912a9c9d7f/vapi/simulator/simulator.go#L345-L347